Code/Tools/TestImpactFramework/Runtime/Code/Source/TestImpactClientSequenceReportSerializer.cpp

```diff
@@ -603,4 +603,254 @@ namespace TestImpact
 
         return stringBuffer.GetString();
     }
+
+    AZStd::chrono::high_resolution_clock::time_point TimePointFromMsInt64(AZ::s64 ms)
+    {
+        return AZStd::chrono::high_resolution_clock::time_point(AZStd::chrono::milliseconds(ms));
+    }
+
+    AZStd::vector<Client::Test> DeserializeTests(const rapidjson::Value& serialTests)
+    {
+        AZStd::vector<Client::Test> tests;
+        tests.reserve(serialTests[SequenceReportFields::Keys[SequenceReportFields::Tests]].GetArray().Size());
+        for (const auto& test : serialTests[SequenceReportFields::Keys[SequenceReportFields::Tests]].GetArray())
+        {
+            const AZStd::string name = test[SequenceReportFields::Keys[SequenceReportFields::Name]].GetString();
+            const auto result = TestResultFromString(test[SequenceReportFields::Keys[SequenceReportFields::Result]].GetString());
+            tests.emplace_back(name, result);
+        }
+
+        return tests;
+    }
+
+    Client::TestRunBase DeserializeTestRunBase(const rapidjson::Value& serialTestRun)
+    {
+        return Client::TestRunBase(
+            serialTestRun[SequenceReportFields::Keys[SequenceReportFields::Name]].GetString(),
+            serialTestRun[SequenceReportFields::Keys[SequenceReportFields::CommandArgs]].GetString(),
+            TimePointFromMsInt64(serialTestRun[SequenceReportFields::Keys[SequenceReportFields::StartTime]].GetInt64()),
+            AZStd::chrono::milliseconds(serialTestRun[SequenceReportFields::Keys[SequenceReportFields::Duration]].GetInt64()),
+            TestRunResultFromString(serialTestRun[SequenceReportFields::Keys[SequenceReportFields::Result]].GetString()));
+    }
+
+    template<typename TestRunType>
+    AZStd::vector<TestRunType> DeserializeTestRuns(const rapidjson::Value& serialTestRuns)
+    {
+        AZStd::vector<TestRunType> testRuns;
+        testRuns.reserve(serialTestRuns.GetArray().Size());
+        for (const auto& testRun : serialTestRuns.GetArray())
+        {
+            testRuns.push_back(TestRunType(DeserializeTestRunBase(testRun)));
```

Comment:
Since this is creating the elemen `TestRunType` in place, could benefit from using `emplace_back`?

Like this `testRuns.emplace_back(DeserializeTestRunBase(testRun));`

Reply:
Good catch, I'll fix 👍